coreDriverInterfaceSupported.json

```diff
@@ -1,4 +1,4 @@
 {
     "_comment": "contains a list of core-driver interfaces branch names that this core supports",
-    "versions": ["2.8", "2.9", "2.10", "2.11", "2.12", "2.13", "2.14", "2.15", "2.16", "2.17", "2.18"]
```

Comment:
To discuss affect on:
- websocket auth
- graphql docs
- using regular jwt lib vs getSession
- m2m auth
- one API being used for m2m and being called from frontend

Reply:
# websocket auth
- cookie based
   - enabled exposeAccessTokenToFrontend
   - on frontend get the access token using session.getAccessTokens
   - send that to websocket connection as part of the message
   - use standard jwt lib in websocket to verify the token pointing to the jwks endpoint
   - session refreshing will not be an issue cause when we call session.getAccessTokens on the frontend, it ensures that the access token is valid.
- header based
   - on frontend get the access token using session.getAccessTokens
   - send that to websocket connection as part of the message
   - use standard jwt lib in websocket to verify the token pointing to the jwks endpoint
   - session refreshing will not be an issue cause when we call session.getAccessTokens on the frontend, it ensures that the access token is valid.

# graphql 
TODO

Reply:
## GraphQL

The existing guide should still work and it's still necessary if we recommend people to use `getSession`. It could be made easier however:
1. By recommending `getSessionWithoutModifyingResponse` and handling setting things in the response themselves.
2. By recommending standard JWT libraries. This will limit how they interface with our sessions, but they only need to handle sending a 401 + the GQL servers generally have guides about JWT auth

No change necessary/recommended on the FE, since regular interception should already take care of everything.

## using regular jwt lib vs getSession

- This will enable people to use standard JWT verification libraries instead of our `getSession` on the backend.
- It can make integration with API gateway and other similar things easier as there are existing guides/built-in solutions for JWT based auth for most.
- These verification libs generally send 401s for invalid tokens, which can make integration even easier
- `exposeAccessTokenToFrontend` makes it possible to use cookie based auth on some (i.e.: the auth domain) and header based on other domains (using `Session.getAccessToken()` and setting it as the auth header manually for other domains.

No change necessary/recommended on the FE, since regular interception should already take care of everything.

## m2m auth

No change - the tokens issued to users in general should not be used in M2M communication. The JWT recipe should be used to create the tokens used in M2M communication, and the JWKs endpoint exposed both by the core and the BE SDK can be used for validation.

## one API being used for m2m and being called from frontend

- Now it can use a single JWT validation middleware
- It then needs to validate where the request came from - e.g.: by checking the sub
- Existing implementations will keep working, `verifySession` can still differentiate the access tokens from jwts created otherwise

## Session sharing across API domains (not just sub-domains)

- Header-based sessions
  - no change
- Cookie-based sessions
  - The auth domain can use cookies after this change
  - This means that the refresh token is still stored securely in an HTTPOnly cookie.
  - Steps:
    - enable `exposeAccessTokenToFrontend`
    - on frontend get the access token using `Session.getAccessToken`
    - attach the access token as an authorization header to requests made to API domains
    - handle 401s coming from API domains by calling `attemptRefreshingSession`

Reply:
Sounds good.